Model/Sync/Reset/Main.php

```diff
@@ -161,8 +186,80 @@ private function deleteRunningCronSchedules()
     public function setResetInProgressConfig($storeId, $flag)
     {
         $yotpoEntityName = $this->getYotpoEntityName();
+        if (!$yotpoEntityName) {
+            return;
+        }
         $key = 'reset_sync_in_progress_' . $yotpoEntityName;
         $this->config->saveConfig($key, $flag, $storeId);
         $this->cacheTypeList->cleanType(\Magento\Framework\App\Cache\Type\Config::TYPE_IDENTIFIER);
     }
+
+    /**
+     * @param string $tableName
+     * @param int $attributeId
+     * @param int $storeId
+     * @return int
+     */
+    public function getCountOfEntities($tableName, $attributeId, $storeId)
+    {
+        $connection  = $this->resourceConnection->getConnection();
+        $tableName = $this->resourceConnection->getTableName($tableName);
+        $select = $connection->select();
+        $query = $select->reset()
+            ->from(
+                ['p' => $tableName]
+            );
+        if ($storeId) {
+            $query->where('store_id = ?', $storeId);
+        }
+        $query->where('attribute_id = ?', $attributeId);
+        $query->where('value = ?', 1);
+        return $connection->query($query)->rowCount();
+    }
+
+    /**
+     * @param array<mixed> $dataSet
+     * @param int $storeId
+     * @return void
+     */
+    public function updateEntityAttributeTableData($dataSet, $storeId)
```

Comment:
This is not exactly what I meant, this still contains a type that is unknown to the reader - `dataSet`.

I'd like the declaration of the method to be something like;
```
public function updateEntityAttributeTableData($storeId, $attributeCode, $tableName).
```

There's no actual meaning to dataSet it this case further than an object to transfer certain data, this can and should be simplified.

Reply:
Okay, we will make these changes